tensorflow-opgen/src/TensorFlow/OpGen/ParsedOp.hs

```diff
@@ -0,0 +1,301 @@
+-- | This module helps parse the proto OpDef into a Haskell type which is more
+-- descriptive of how the attributes and arguments will be used in the
+-- generated code.
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+module TensorFlow.OpGen.ParsedOp
+    ( ParsedOp(..)
+    , Name(..)
+    , HaskellName(..)
+    , TFName(..)
+    , Attr(..)
+    , AttrType(..)
+    , AttrBaseType(..)
+    , ParsedArg(..)
+    , ParsedArgCase(..)
+    , ArgType(..)
+    , ArgKind(..)
+    , parseOp
+    , camelCase
```

Comment:
It shared enough code with ParsedOp (via `forceCase`) that I felt it was less awkward to just define here.